python/ray/dataframe/indexing.py

```diff
@@ -1,104 +1,431 @@
 import ray
-from .dataframe import _deploy_func
+from warnings import warn
+
+from pandas.api.types import (is_scalar, is_list_like, is_bool)
+from pandas.core.dtypes.common import is_integer
+
+from .utils import (_blocks_to_col, _get_nan_block_id, extractor)
+from .index_metadata import _IndexMetadata
+
+
+def is_slice(x):
+    return isinstance(x, slice)
+
+
+def is_2d(x):
+    return is_list_like(x) or is_slice(x)
+
+
+def is_tuple(x):
+    return isinstance(x, tuple)
 
 
 class _Location_Indexer_Base():
     """Base class for location indexer like loc and iloc
-    This class abstract away commonly used method
     """
 
     def __init__(self, ray_df):
         self.df = ray_df
+        self.col_coord_df = ray_df._col_metadata._coord_df
+        self.row_coord_df = ray_df._row_metadata._coord_df
+        self.block_oids = ray_df._block_partitions
 
     def __getitem__(self, key):
+        # Pathological Case: df.loc[1:,]
+        if is_tuple(key) and len(key) == 1:
+            key = key[0]
+
+        # The one argument case is equivalent to full slice in 2nd dim.
+        # iloc[3] == iloc[3, :] for a dataframe
         if not isinstance(key, tuple):
-            # The one argument case is equivalent to full slice in 2nd dim.
-            return self.locate_2d(key, slice(None))
-        else:
-            return self.locate_2d(*key)
-
-    def _get_lookup_dict(self, ray_partition_idx):
-        if ray_partition_idx.ndim == 1:  # Single row matched
-            position = (ray_partition_idx['partition'],
-                        ray_partition_idx['index_within_partition'])
-            rows_to_lookup = {position[0]: [position[1]]}
-        if ray_partition_idx.ndim == 2:  # Multiple rows matched
-            # We copy ray_partition_idx because it allows us to
-            # do groupby. This might not be the most efficient method.
-            # And have room to optimize.
-            ray_partition_idx = ray_partition_idx.copy()
-            rows_to_lookup = ray_partition_idx.groupby('partition').aggregate(
-                lambda x: list(x)).to_dict()['index_within_partition']
-        return rows_to_lookup
-
-    def locate_2d(self, row_label, col_label):
+            return self._triage_getitem(key, slice(None))
+
+        return self._triage_getitem(*key)
+
+    def _triage_getitem(self, row_loc, col_loc):
+        # iloc[1,2] returns a scaler
+        if is_scalar(row_loc) and is_scalar(col_loc):
+            return self._get_scaler(row_loc, col_loc)
+
+        # iloc[1,:] returns a row series: row 1
+        if is_scalar(row_loc) and is_2d(col_loc):
+            return self._get_row_series(row_loc, col_loc)
+
+        # iloc[:,1] returns a columns series: col 1
+        if is_2d(row_loc) and is_scalar(col_loc):
+            return self._get_col_series(row_loc, col_loc)
+
+        if is_2d(row_loc) and is_2d(col_loc):
+            return self._get_dataframe_view(row_loc, col_loc)
+
+    def _get_scaler(self, row_loc, col_loc):
```

Comment:
Are you planning to implement these? It would be better to have a `NotImplementedError` than just `pass`. That way we don't have some silent internal error or something.